bumble/device.py

```diff
@@ -730,12 +749,12 @@ def __init__(self):
         )
         self.irk = bytes(16)  # This really must be changed for any level of security
         self.keystore = None
-        self.gatt_services = []
+        self.gatt_services: List[Dict[str, Any]] = []
 
-    def load_from_dict(self, config):
+    def load_from_dict(self, config: Dict[str, Any]) -> None:
         # Load simple properties
         self.name = config.get('name', self.name)
-        self.address = Address(config.get('address', self.address))
+        self.address = Address(config.get('address', DEVICE_DEFAULT_ADDRESS))
```

Comment:
The intent of the original code was to only change address if specified in the dict, else leave it unchanged (i.e don't force it to DEVICE_DEFAULT_ADDRESS).